### ISSUE TITLE
Compile error when emphasis is used both inside and outside of inline note in same paragraph

### ISSUE THREAD

**Issue by nopria** [Witiko/markdown#447]
The following MWE doesn't compile "Example 2" line.
If you comment such line it compiles regularly.

```latex
\documentclass{article}
\usepackage[hybrid, inlineNotes=true]{markdown}

\begin{document}
\begin{markdown}

# MWE emphasis in footnotes with markdown

Example 1 ^[Example *starred*.].

Example 2 with *more* starred ^[Example *starred*.].

\end{markdown}
\end{document}
```

This issue was not present in some prevoius version (don't know the exact number but I know I was using option `inlineFootnotes=true` instead of `inlineNotes=true` as required by current version).

**Comment by Witiko** [comment]
Hi and sorry to hear about your trouble. Which version of the Markdown package are you using and what happens when the document "doesn't compile"? If you could attach your `.log` file, that should answer both my questions.

**Comment by nopria** [comment]
Sure, here it is the log file:

[test_emph_in_footnote.log](https://github.com/user-attachments/files/15780257/test_emph_in_footnote.log)

**Comment by Witiko** [comment]
According to the log, you use the latest released version v3.6.0 (2024-05-27) of the Markdown package with XeLaTeX from TeX Live 2024 and the error is as follows:

```
! Missing $ inserted.
<inserted text> 
                $
l.13 \end{markdown}
                   
? x
```

I tried compiling the document in the same conditions and I successfully reproduced your issue. Enabling the `eagerCache = true` made the error more obvious:

```
$ xelatex --shell-escape example.tex
! Missing $ inserted.
<inserted text> 
                $
l.5 ...th \markdownRendererEmphasis{more} starred^
                                                  [Example \markdownRenderer...

? x
```

Your second inline note is not parsed as inline notes and the character `^` is taken literally by TeX as a superscript character. Here is how your document is parsed:

```
$ cat ./_markdown_example/8838be0c1fda1485b6ded8c982bd70ba.md.tex
```
``` tex
\markdownRendererDocumentBegin
\markdownRendererSectionBegin
\markdownRendererHeadingOne{MWE emphasis in footnotes with markdown}\markdownRendererInterblockSeparator
{}Example 1\markdownRendererNote{Example \markdownRendererEmphasis{starred}.}.\markdownRendererParagraphSeparator
{}Example 2 with \markdownRendererEmphasis{more} starred^[Example \markdownRendererEmphasis{starred}.].
\markdownRendererSectionEnd \markdownRendererDocumentEnd
```

@lostenderman Would you mind looking at why this fails? It seems to do with how we process emphasis; if I replace `Example 2 with *more* starred` with `Example 2 with more starred` or `Example 2`, the second inline note is correctly parsed.

**Comment by nopria** [comment]
I don't think the bug is related only to the way emphasis it's parsed, in fact from more tests I saw that the same bug shows up with bold too and actually **any combination of bold/emphasis inside AND outside the footnote** does NOT compile:

```latex
\documentclass{article}
\usepackage[hybrid, inlineNotes=true]{markdown}

\begin{document}
\begin{markdown}

# MWE emphasis in footnotes with markdown

Combinations of emphasis/bold inside OR outside footnote compile correctly:

1. Example *emphasis* bold outside footnote^[Example emphasis bold inside footnote.].
1. Example emphasis bold outside footnote^[Example *emphasis* bold inside footnote.].
1. Example emphasis **bold** outside footnote^[Example emphasis bold inside footnote.].
1. Example emphasis bold outside footnote^[Example emphasis **bold** inside footnote.].

% Any combination of emphasis/bold inside AND outside footnote does NOT compile
%1. Example *emphasis* bold outside footnote^[Example *emphasis* bold inside footnote.].
%1. Example emphasis **bold** outside footnote^[Example emphasis **bold** inside footnote.].
%1. Example *emphasis* bold outside footnote^[Example emphasis **bold** inside footnote.].
%1. Example emphasis **bold** outside footnote^[Example *emphasis* bold inside footnote.].

\end{markdown}
\end{document}
```

It might be something related to footnotes.

N.B. Mixed markdown/latex combinations of emphasis/bold inside AND outside footnote compiles correctly.

**Comment by nopria** [comment]
An additional detail, the issue shows up only if emphasis/bold comes **before** footnote containing other emphasis/bold.
If emphasis/bold follows a footnote containing emhasis/strong, the source compile regularly.

**Comment by Witiko** [comment]
@nopria: The issue has been fixed by @lostenderman in #450, see also your example document and its output after #450:

``` tex
\documentclass{article}
\usepackage[hybrid, inlineNotes]{markdown}

\begin{document}
\begin{markdown}

# MWE emphasis in footnotes with markdown

Combinations of emphasis/bold inside OR outside footnote compile correctly:

1. Example *emphasis* bold outside footnote^[Example emphasis bold inside footnote.].
1. Example emphasis bold outside footnote^[Example *emphasis* bold inside footnote.].
1. Example emphasis **bold** outside footnote^[Example emphasis bold inside footnote.].
1. Example emphasis bold outside footnote^[Example emphasis **bold** inside footnote.].

Any combinations of emphasis/bold inside AND outside footnote also compile correctly since #450:

1. Example *emphasis* bold outside footnote^[Example *emphasis* bold inside footnote.].
1. Example emphasis **bold** outside footnote^[Example emphasis **bold** inside footnote.].
1. Example *emphasis* bold outside footnote^[Example emphasis **bold** inside footnote.].
1. Example emphasis **bold** outside footnote^[Example *emphasis* bold inside footnote.].

\end{markdown}
\end{document}
```
![image](https://github.com/Witiko/markdown/assets/603082/fd781edf-3112-4943-bd8c-72f6dff04a43)
![image](https://github.com/Witiko/markdown/assets/603082/bac16947-6a67-4360-830a-eb6bd5ce10a1)

I also added a [regression test][2], so that any similar issues are caught by our automated tests in the future.

The fix will be part of Markdown 3.7.0, to be released later this month. If you'd like to use the fix earlier, take the following steps:

1. Uninstall the portable installation of version 2.23.0 from https://github.com/Witiko/markdown/discussions/449.
2. Download [the archive `markdown.zip` for the latest prerelease of the Markdown package][1] and extract the file `markdown.lua`.
3. Place the file `markdown.lua` to the directory next to your document.

Note that leaving the file `markdown.lua` in the directory next to your document will eventually cause trouble due to the version mismatch of the file `markdown.lua` and the rest of the Markdown package from TeX Live. However, it is a good interim solution before Markdown 3.7.0 has been released.

Even after Markdown 3.7.0, using an old file `markdown.lua` should only prevent you from using new features but should be otherwise fully forward-compatible; at least until Markdown 4, which may change the interface between `markdown.lua` and the rest of the Markdown package. However, there are currently no plans for Markdown 4 and I don't imagine there would be, at least in the next couple years.

 [1]: https://github.com/Witiko/markdown/releases/download/latest/markdown.zip
 [2]: https://github.com/Witiko/markdown/blob/dbd5f3cbb553dd8d97c67ae0ae1cfd9e390e9457/tests/testfiles/regression/github/issue-447.test

**Comment by Witiko** [comment]
I released the new version yesterday. In the end, it was not a minor release 3.7.0 but just a bugfix release 3.6.1, since no new features were added. In you have a TeX Live install that is not locked by your distro to prevent updates, you should be able to update the Markdown package using the command `(sudo) tlmgr update markdown`.
